### DISCUSSION TITLE
Does the executable produced by bun build --compile still depend on external files?

### DISCUSSION THREAD

**Discussion by guest271314** [oven-sh/bun#14649] [Q&A]
I compiled `npm` to a standalone executable with `bun`.

I deleted the source `npm` package files after the compilation, then copied the executable to a different disk, then copied the executable back to the original directory.

I ran the executable again, and it threw, citing the standalone executable can't find a module in the deleted folder

```
bun-npm --help
error: Cannot find module "/home/user/bin/node-v23/lib/node_modules/npm/lib/cli/entry.js" from "/$bunfs/root/bun-npm"
      at syntaxErrorHandler (/$bunfs/root/bun-npm:286:16)
```

**Comment by guest271314** [comment]
I think this is a bug https://github.com/oven-sh/bun/issues/14676.

**Comment by juliankrispel** [comment]
same here. It's not a standalone executable if it doesn't actually work like advertised
